### PR TITLE
feat(summary): AZ Func weekly task owner report worker

### PR DESCRIPTION
**Pull request by Jonathanio123**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**

[AB#57451](https://statoil-proview.visualstudio.com/787035c2-8cf2-4d73-a83e-bb0e6d937eec/_workitems/edit/57451)

Similar to resource owners report:

- This worker retrieves relevant data to create the report metrics and then stores it in summary api

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

Written tests for the report mertrics. Done simple tests for 2 projects

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

**Review by aleklundeq — APPROVED** [review]

**Comment by Jonathanio123** [comment]
Waiting to merge this until I've verified the data calculation with PO

File: src/Fusion.Summary.Functions/ReportCreator/WeeklyTaskOwnerReportDataCreator.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            // Handle case where the position is not currently active
+            if (activeInstance is null)
+            {
+                if (futureInstances.Count == 0)
+                    continue; // This is a past position
+
+                var endingPositionAllocation = FindFirstTBNOrLastExpiringInstance(futureInstances);
+
+                // If the last instance is not the last instance then there are more instances after it that are not within the 3-month window or TBN
+                var isEndingInstanceLast = futureInstances.Last() == endingPositionAllocation;
```
Comment (aleklundeq) [inline src/Fusion.Summary.Functions/ReportCreator/WeeklyTaskOwnerReportDataCreator.cs:106]: A bit nitpick, but the comment related to this is some what unclear.
But it also makes me think about the cases where we have several instances that starts after today and ends within 3 months where some are allocated, and some are not. This should be an edge case of course.

Comment (Jonathanio123) [reply src/Fusion.Summary.Functions/ReportCreator/WeeklyTaskOwnerReportDataCreator.cs:111]: Updated comment and added another test
